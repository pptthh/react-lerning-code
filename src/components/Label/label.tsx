import * as React from 'react';
import { NOOP } from '../../utils';

export interface ILabel {
    elementClass?: string;
    tooltip?: string;
    children: string;
    onClick?: React.MouseEventHandler<HTMLSpanElement>;
}

// fixme: check props for valid text!!
const Label = ({onClick = NOOP, tooltip = '', children, elementClass = ''}: ILabel): JSX.Element =>
    <span className={elementClass} title={tooltip} onClick={onClick}>{children}</span>;

export default Label;
