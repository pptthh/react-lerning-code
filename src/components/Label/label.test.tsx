import { configure, render, shallow } from 'enzyme';
import * as ReactSixteenAdapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import Label from './label';

configure({ adapter: new ReactSixteenAdapter() });

describe('describe Label', () => {
    it('renders correctly with Text only', () => {
        const tree = render(<Label>simple text22</Label>);

        expect(toJson(tree)).toMatchSnapshot();
    });

    it('shallow renders correctly with params', () => {
        let clickCounter = 0;
        const tree = shallow(
            <Label tooltip={'tooltip'} className={'elementClass'} onClick={() => clickCounter++}>
                renders correctly with params
            </Label>);

        expect(toJson(tree)).toMatchSnapshot();
        tree.simulate('click');
        expect(clickCounter).toEqual(1);
    });

    it('Shallow render example', () => {
        const tree = shallow(<h1><Label>simple text22</Label></h1>);

        expect(toJson(tree)).toMatchSnapshot();
    });
});
