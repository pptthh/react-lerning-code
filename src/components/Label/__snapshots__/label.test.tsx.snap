// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`describe Label Shallow render example 1`] = `
<h1>
  <Label>
    simple text22
  </Label>
</h1>
`;

exports[`describe Label renders correctly with Text only 1`] = `
<span
  class=""
  title=""
>
  simple text22
</span>
`;

exports[`describe Label shallow renders correctly with params 1`] = `
<span
  className="elementClass"
  onClick={[Function]}
  title="tooltip"
>
  renders correctly with params
</span>
`;
